File: Objectivec/Day5/webServiceDeom/webServiceDeom/DetailsViewController.m
//
//  DetailsViewController.m
//  webServiceDeom
//
//  Created by Esraa Hassan on 3/12/20.
//  Copyright © 2020 Yahya. All rights reserved.
//

#import "DetailsViewController.h"

@interface DetailsViewController ()

@end

@implementation DetailsViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    _nameLabel.text = _colleague.name;
    _phoneLabel.text = _colleague.phone;
    [_imageView setImage:_image];
    
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}



- (IBAction)cancelButtonAction:(id)sender {
    [self.presentingViewController dismissViewControllerAnimated:YES completion:nil];
}
@end

File: Practice/MoviesPracticeMVP/MoviesPracticeMVP/Features/Details/View/DetailsViewController.swift
//
//  ViewController.swift
//  MovieList
//
//  Created by Esraa Hassan on 3/25/20.
//  Copyright © 2020 Yahya. All rights reserved.
//

import UIKit
import Kingfisher
class DetailsViewController: UIViewController {
    

    @IBOutlet weak var detailsScrollView: UIScrollView!
    
    @IBOutlet weak var ratingLabel: UILabel!
    @IBOutlet weak var nameLabel: UILabel!
    
    @IBOutlet weak var releaseYearLabel: UILabel!
    
    @IBOutlet weak var genreLabel: UILabel!
    
    @IBOutlet weak var movieImageView: UIImageView!
    
    var movie: Movie?
  

    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        nameLabel.text = movie!.title
        detailsScrollView.contentSize = CGSize(width: 100   , height:   100)
        if movie!.image.contains("http"){
            movieImageView.kf.setImage(with:URL(string: movie!.image))
        }else{
            var imageURL = Constants.imageBaseURL
            imageURL.append(movie!.image)
            
            movieImageView.kf.setImage(with: URL(string: imageURL), placeholder: UIImage(named: "noImage.png"), options: [], progressBlock: nil, completionHandler: nil)
        }
        genreLabel.text = movie!.genre
        
        releaseYearLabel.text = String(movie!.releaseYear)
        ratingLabel.text = String(movie!.rating)
       
    }

  
    
   
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        
    }


}

